security/login_link.rst

```diff
@@ -0,0 +1,510 @@
+
+Now that the authenticator is able to check the login links, you must
+create a page where a user can request a login link and login to your
+website.
+
+The login link can be generated using the
+:class:`Symfony\\Component\\Security\\Http\\LoginLink\\LoginLinkHandlerInterface`.
+The correct login link handler is autowired for you when type-hinting for
+this interface::
+
+    // src/Controller/SecurityController.php
+    namespace App\Controller;
+
+    use App\Repository\UserRepository;
+    use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+    use Symfony\Component\HttpFoundation\Request;
+    use Symfony\Component\Routing\Annotation\Route;
+    use Symfony\Component\Security\Http\LoginLink\LoginLinkHandlerInterface;
+
+    class SecurityController extends AbstractController
+    {
+        /**
+         * @Route("/login", name="login")
+         */
+        public function requestLoginLink(LoginLinkHandlerInterface $loginLinkHandler, UserRepository $userRepository, Request $request)
+        {
+            // check if login form is submitted
+            if ($request->isMethod('POST')) {
+                // load the user in some way (e.g. using the form input)
+                $email = $request->request->get('email');
+                $user = $userRepository->findOneBy(['email' => $email]);
+
+                // create a login link for $user
+                $loginLinkDetails = $loginLinkHandler->createLoginLink($user);
+                $loginLink = $loginLinkDetails->getUrl();
+
+                // ... send the $loginLink (e.g. using the Mailer or Notifier component)
+
+                // render a "Login link is sent!" page
+                return $this->render('security/login_link_sent.html.twig');
+            }
+
+            // if it's not submitted, render the "login" form
+            return $this->render('security/login.html.twig');
+        }
+    }
+
+.. code-block:: html+twig
+
+    {# templates/security/login.html.twig #}
+    {% extends 'base.html.twig' %}
+
+    {% block body %}
+    <form action="{{ path('login') }}" method="POST">
+        <input type="email" name="email">
+        <button type="submit">Login</button>
+    </form>
+    {% endblock %}
+
+In this controller, the user is submitting their e-mail address to the
+controller. Based on this property, the correct user is loaded and a login
+link is created using
+:method:`Symfony\\Component\\Security\\Http\\LoginLink\\LoginLinkHandlerInterface::createLoginLink`.
+
+.. caution::
+
+    It is important to send this link to the user and **not show it directly**,
+    as that would allow anyone to login. For instance, use the
+    :doc:`mailer </mailer>` component to mail the login link to the user.
+    Or use the :doc:`notifier </notifier>` component to send an SMS to the
+    user's device.
+
+.. note::
+
+    The service or controller using the ``LoginLinkHandlerInterface`` must
+    be used only if the current request is covered by a
```

Comment:
Side question: is this validated/checked/asserted in the request flow? Feel weird to document such case, or maybe use a caution instead of note?

Reply:
You'll get an error "No login link handler found as the current route is not covered by a firewall.".

I'm not sure if we should warn users about this in the docs, or if they should just encounter the error and then find out what they were doing wrong.

Reply:
I'm also not sure about this warning. If you're building multiple firewalls, you're already disregarding the "normal" advice.

Reply:
Alright, let's remove it then.